tests/testmacro.nim

```diff
@@ -136,6 +136,27 @@ suite "Macro transformations test suite":
     check:
       waitFor(gen(int)) == default(int)
 
+  test "Nested return":
+    proc nr: Future[int] {.async.} =
+      return
+        if 1 == 1:
+          return 42
+        else:
+          33
+
+    check waitFor(nr()) == 42
+
+  test "Issue #415 (run closure to completion on return)":
+    var x = 0
+    proc test415 {.async.} =
+      try:
+        return
+      finally:
+        await stepsAsync(1)
```

Comment:
I want to see more than one `await` in `finally`. and also i want to see how it works in exception handler, e.g. more than one `await`. Also we need to test `defer` functionality.